zenml/core/steps/tokenizer/base_tokenizer.py

```diff
@@ -0,0 +1,48 @@
+﻿#  Copyright (c) maiot GmbH 2021. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+from typing import Text, List
+from zenml.core.steps.base_step import BaseStep
+
+
+class BaseTokenizer(BaseStep):
+    """
+    Base step class for all tokenizers.
+    """
+
+    def __init__(self,
+                 text_feature: Text,
+                 skip_training: bool = False,
+                 **kwargs):
+        super(BaseTokenizer, self).__init__(text_feature=text_feature,
+                                            skip_training=skip_training,
+                                            **kwargs)
+
+        self.text_feature = text_feature
+        self.skip_training = skip_training
+
+    def train(self, files: List[Text]):
+        raise NotImplementedError
+
+    def train_from_iterator(self, files: List[Text]):
```

Comment:
What is the difference between `train` and `train_from_iterator` in this interface? They both seem to do the same things. We can just keep the `train` to make it less complex

Reply:
Good catch! The distinction comes from Huggingface, which has both a `train` and a `train_from_iterator`, but that should not influence the BaseTokenizer's API design

Reply:
Interface cleaned up in 6d8a3e5